.github/workflows/python-app.yml

```diff
@@ -0,0 +1,47 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: gitchangelog
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - name: Git repository creation (required)
+      run: |
+        git init
+        git fetch
+        git reset origin/main
+        git checkout origin/main
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install wheel
+        pip install minimock mako pystache
+        pip install flake8 pytest
```

Comment:
I guess we can have all in one `pip3 install` command? Or, even better, have those in a `requirements-test.txt` file?

And I'd love to see caching for pip: https://medium.com/ai2-blog/python-caching-in-github-actions-e9452698e98d

Reply:
:+1: Agree. Lets do it in followup PR as proposed